### ISSUE TITLE
MASP parameters should be included in Anoma releases created by `anomac utils init-network`

### ISSUE THREAD

**Issue by james-chf** [anoma/namada#100]
We should bundle the MASP `.params` files directly into the Anoma .tar.gzs that are distributed to be downloaded by `anomac utils join-network`.

When running `anomac utils init-network` to generate the .tar.gz, we should include details of MASP parameter files in the network config file, and hence the genesis file that is generated from that, similar to how we include details of some precompiled WASMs.

e.g.
```toml
[masp_params.output]
filename = "masp-output.params"
sha256 = "8c9bb99a9eb45bdb62cda38b2d09b45eda22138910db5b194aa8c1ba1fdfda78"  # hash

# ...
```

These MASP params will be stored in a `masp/` directory in the chain directory.
```
Permissions Size User Date Modified Name
drwxr-xr-x     - main 13 Apr 14:48  .anoma
drwxr-xr-x     - main 13 Apr 14:49  ├── anoma-masp-0.3.51d2f83a8412b95
drwxr-xr-x     - main 13 Apr 14:49  │  ├── masp
.rw-r--r--   15M main 24 Mar 08:25  │  │  ├── masp-output.params
.rw-r--r--   48M main 24 Mar 08:25  │  │  └── masp-spend.params
```

Eventually `anoma client` should start using MASP params from this directory rather than the default implicit directory (i.e. `~/Library/Application Support/MASPParams` or `~/.masp-params`)

**Comment by cwgoes** [comment]
@mariari can you check on this one?

**Comment by mariari** [comment]
Had a discussion about this issue with @tzemanovic, it is best to wait on this issue until both #53 and #685  are finished, from there we would have a global base dir (`~/.local/share/namada`) in which we can save the `masp-params` by their hash.

Further, the issue is a bit out of date, the `namadac utils init-network` command creates `tar.gz` files that are around 3.61 kilobytes, containing:

```
.namada/anoma-feigenbaum-0.19107c4f181.toml
.namada/global-config.toml
.namada/anoma-feigenbaum-0.19107c4f181/config.toml
.namada/anoma-feigenbaum-0.19107c4f181/wasm/checksums.json
```

while the `join-network` command grabs the following files

```bash
.
├── global-config.toml
├── public-testnet-2.1.9088a928d0c
│   ├── config.toml
│   └── wasm
│       ├── checksums.json
│       ├── tx_bond.50cc7754f07b0553d22d2f259b7be0d0dfa83cac21f349b9a5ef0c7f62a4c8e3.wasm
│       ├── tx_change_validator_commission.29944a5b1a31077c0d5e486c30338f9a46ff71daa91fc3619b8425c0a0592203.wasm
│       ├── tx_ibc.4d8d041cea7a9fd165ad082ba24dcb1afc8591a0fafdf36b6729a2cf8389d001.wasm
│       ├── tx_init_account.604050807200b80fb6cccaf49e62cd43ab48a6fcc03fdec5c8148e8d01067117.wasm
│       ├── tx_init_proposal.f84c91b5306664c054f2924869f407d42cc39e7a268e9d418a631e89dd7acbf4.wasm
│       ├── tx_init_validator.c03ffe483d285866f596fc38a6fc5df59dc1ce0a45d3b6fde9fd597552284a0b.wasm
│       ├── tx_reveal_pk.4e0f779ed1600e7208620233cb64b93fc56afb81122dd3f9bd5cf76dc3beff94.wasm
│       ├── tx_transfer.8c4fccac307878e20df2b4e18f418eea2bd7b0c0cb70d4e97fdc861d3deb2b76.wasm
│       ├── tx_unbond.879f36b2fe0f56f5f8599a02a4716ee296037abfe2c384bee55809976918feca.wasm
│       ├── tx_update_vp.db4d8c11658c5f8e99fc39f0112be1ee480ab1f0045010d323ee3081a7afe802.wasm
│       ├── tx_vote_proposal.16fb6a28f46a2d3d6013f4b7b20d8fe556cdd11b38f9bf064b085b266e0b54cc.wasm
│       ├── tx_withdraw.90cb4a92c2ffaf93dbb6825667aec8a4cb571f0f2c9dd4e0592590a289416011.wasm
│       ├── vp_implicit.50be0b491aa3c8325acba7af94ad2abb709d79c7e56d6b78e0dcf88a1750e65b.wasm
│       ├── vp_masp.403762ff0878a453cb26298d6b6fce525e0482a6441cdf4d9738af9d7b3a7534.wasm
│       ├── vp_testnet_faucet.80d79f6565cd18e1a7791df39d4537582380d21cadc071d8a5876a3adf196715.wasm
│       ├── vp_token.30b4833f66d244699bfa1e7ef7fb75bae3dae532b7361f9e2da3ea9492512960.wasm
│       ├── vp_user.0d74f57d2aa43bcc81f4bb5db474d6ba0bd006944de06d5245791ac898835df1.wasm
│       └── vp_validator.7433e3e52fcd72ec50c79171f06b5b38f149209a9ed6d96df6c8e098c530932b.wasm
└── public-testnet-2.1.9088a928d0c.toml

3 directories, 22 files

```

which produce a total zip size of over 2 mega bytes. Meaning that due to size constraints the `wasm` files are not included in the bundle. If we look at the masp parameters they add up to around 80 megabytes and for the same argument should not be included in the zips.

However, the hash of the masp parameters should be included in the namada release

**Comment by cwgoes** [comment]
Out-of-date / superseded by more recent work.
